rest/data/host.go

```diff
@@ -467,7 +459,10 @@ func (hc *MockConnector) AggregateSpawnhostData() (*host.SpawnHostUsage, error)
 func findHostByIdWithOwner(c Connector, hostID string, user gimlet.User) (*host.Host, error) {
 	host, err := c.FindHostById(hostID)
 	if host == nil {
-		return nil, err
+		return nil, gimlet.ErrorResponse{
+			StatusCode: http.StatusNotFound,
+			Message:    fmt.Sprintf("host with id '%s' not found", hostID),
+		}
 	}
 
 	if err != nil {
```

Comment:
We should check error first now, since in that case host will be nil but something else may have gone down

Reply:
done